Macro feature

pawel-wozniak opened the pull request.

pawel-wozniak committed:
Macro feature

kowallus commented on src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java:
```diff
@@ -0,0 +1,76 @@
+				.orElseGet(JCheckBoxMenuItem::new);
+	}
+
+	public static void register() {
```
Why is it public? 
The name is a little bit confusing.

pawel-wozniak replied:
Changed to private and renamed

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java:
```diff
@@ -0,0 +1,66 @@
+
+	@Override
+	public void update() {
+		Job2dDriver newDriver = DriverFeature.getDriverManager().getCurrentDriver();
```
Unnecessary dependency with DriverFeature.

pawel-wozniak replied:
Dependency removed

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java:
```diff
@@ -0,0 +1,66 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordingDriverDecorator implements Job2dDriver, Subscriber {
```
SRP rule. Could you separate Subscriber from Job2dDriver?

pawel-wozniak replied:
Moved Subscriber logic to separate class

kowallus reviewed (commented):
Please do some code cleaning.

pawel-wozniak committed:
Macro Feature - refactor and cleanup

Remove dependency to DriverFeature
Separate Subscriber from RecordingDriverDecorator to separate class
Rename register method in RecordingFeature
